plugins/table/plugin.js

```diff
@@ -30,8 +30,12 @@ CKEDITOR.plugins.add( 'table', {
 					element: 'table',
 					right: function( element ) {
 						if ( element.styles ) {
-							if ( element.styles.border && element.styles.border.match( /solid/ ) ) {
-								element.attributes.border = 1;
+							if ( element.styles.border ) {
+								var parsedStyle = CKEDITOR.tools.style.parse.border( element.styles.border );
+								if ( parsedStyle.style && parsedStyle.style === 'solid' &&
```

Comment:
this is temporary solution until #619 will be done. HTML border attribute is for solid border that's why we check only if its set to solid. It's the same behaviour as it was, just fixed for zero width issue.